app/backend/src/tests/test_fixtures.py

```diff
@@ -38,8 +38,11 @@ def db():
     """
     engine = create_engine(config["DATABASE_CONNECTION_STRING"], poolclass=NullPool)
 
-    Base.metadata.drop_all(engine)
-    Base.metadata.create_all(engine)
+    # drop everything currently in the database
+    engine.execute("DROP SCHEMA public CASCADE; CREATE SCHEMA public;")
+
+    # rebuild it with alembic migrations
+    apply_migrations()
```

Comment:
Please write separate tests for migrations instead of making it part of every little unit test.

Reply:
Why? I think it'd be good to try it against whatever schema the migrations build, this is closer to the authentic running environment.

Another option might be if we could get a very strict check that would build the DB with migrations and then compare to the `create_all` schema; but I think that'd be spottier than just using the generated DB?

According to CI, it seems this is only about 5% slower than with `create_all`.

Reply:
All unit tests should run with the latest schema.
A migration test could start with the earliest supported schema, apply all migrations, and check that the resulting schema is equal to the latest. (maybe alembic provides test methods already?)

It is not about running time, it is about being able to tell what is wrong based on the name of the failing test.

Reply:
Googling around a little I found [alembic-autogen-check](https://github.com/4Catalyzer/alembic-autogen-check). Blindly without knowing what I am doing I ran it on this branch:
```
(venv) chn@rosa:~/couchers/couchers/app/backend$ DATABASE_CONNECTION_STRING=postgresql:///test?host=/home/chn/couchers/temporary-pgsql/datadir PYTHONPATH=src alembic-autogen-check  
[...]
alembic.util.exc.CommandError: Target database is not up to date.
```